shark-android/src/test/java/shark/AndroidReferenceMatchersTest.kt

```diff
@@ -0,0 +1,147 @@
+package shark
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.io.File
+import javax.print.attribute.standard.PrinterMoreInfoManufacturer
+import shark.AndroidReferenceMatchers.Companion.HMD_GLOBAL
+import shark.AndroidReferenceMatchers.Companion.HUAWEI
+import shark.AndroidReferenceMatchers.Companion.INFINIX
+import shark.AndroidReferenceMatchers.Companion.LENOVO
+import shark.AndroidReferenceMatchers.Companion.LG
+import shark.AndroidReferenceMatchers.Companion.MEIZU
+import shark.AndroidReferenceMatchers.Companion.MOTOROLA
+import shark.AndroidReferenceMatchers.Companion.NVIDIA
+import shark.AndroidReferenceMatchers.Companion.ONE_PLUS
+import shark.AndroidReferenceMatchers.Companion.RAZER
+import shark.AndroidReferenceMatchers.Companion.SAMSUNG
+import shark.AndroidReferenceMatchers.Companion.SHARP
+import shark.AndroidReferenceMatchers.Companion.VIVO
+import shark.AndroidReferenceMatchers.Companion.XIAOMI
+import shark.HprofHeapGraph.Companion.openHeapGraph
+import shark.ReferencePattern.StaticFieldPattern
+import shark.ValueHolder.IntHolder
+
```

Comment:
A start on testing for known library leaks.  I didn't test too many edge cases, but enough to ensure the lower API bound is 30 and that the 4 manufacturers we know about do show up as known as expected, and I tried with one that isn't in the list to ensure that it outputs an application leak.